torch/csrc/jit/tensorexpr/exceptions.h

```diff
@@ -26,61 +26,73 @@ namespace torch {
 namespace jit {
 namespace tensorexpr {
 
-class unsupported_dtype : public std::runtime_error {
+class compilation_error : public c10::Error {
  public:
-  explicit unsupported_dtype() : std::runtime_error("UNSUPPORTED DTYPE") {}
+  explicit compilation_error(const std::string& err)
+      : c10::Error(
+            {
+                __func__,
+                __FILE__,
+                static_cast<uint32_t>(__LINE__),
+            },
+            err) {}
```

Comment:
Actually, now that you have made this a single base class from which all the classes below derive, if you can replace `err` with `buildErrorMessage(err)`, that will take care of adding the fuser turn off info as well.